drivers/clk/ralink/Makefile

```diff
@@ -1,2 +1,3 @@
 # SPDX-License-Identifier: GPL-2.0
 obj-$(CONFIG_CLK_MT7621) += clk-mt7621.o
+obj-$(CONFIG_CLK_RALINK) += clk-ralink.o
```

Comment:
You should actually move clk-mt7621 to drivers/clk/mediatek and then add the new mtmips clock driver.

Reply:
I prefer to don't mix things yet. I took a look to the mediatek clock space before I finally put the mt7621 in ralink folder two years ago. MEdiatek's clock code make use of tons of common code that is used along the new arm64 based clock drivers so there was not a good fix since mtmips and ralink datasheets and so on does not explain at all a good clock plan to be able to properly use mediatek's common code. So I think we should first put all the code inside this ralink folder and when all the stuff is accepted think again in this possibility.

Reply:
You don't need to use the MediaTek common clock driver. Just move the clock drivers for MTMIPS SoCs to mediatek directory, just like what I did for pinctrl. The idea is to to put these drivers under mediatek directories as the SoCs are of MediaTek.

```
diff --git a/drivers/clk/mediatek/Kconfig b/drivers/clk/mediatek/Kconfig
index 2d14855dd37e..d29e6df97435 100644
--- a/drivers/clk/mediatek/Kconfig
+++ b/drivers/clk/mediatek/Kconfig
@@ -3,7 +3,22 @@
 # MediaTek Clock Drivers
 #
 menu "Clock driver for MediaTek SoC"
-	depends on ARCH_MEDIATEK || COMPILE_TEST
+	depends on ARCH_MEDIATEK || RALINK || COMPILE_TEST
+
+config CLK_MT7621
+	bool "Clock driver for MediaTek MT7621"
+	depends on SOC_MT7621 || COMPILE_TEST
+	default SOC_MT7621
+	select MFD_SYSCON
+	help
+	  This driver supports MediaTek MT7621 basic clocks.
+
+config CLK_RALINK
+	bool "Clock driver for MTMIPS SoCs"
+	depends on SOC_RT305X || SOC_RT288X || SOC_RT3883 || SOC_MT7620 || COMPILE_TEST
+	select MFD_SYSCON
+	help
+	  This driver supports MTMIPS basic clocks.
 
 config COMMON_CLK_MEDIATEK
 	tristate
```

Reply:
I got your point, thanks. I will make this movement in different patch series to let the reviewers to focus in current aim of this patchset which is the arch clock and reset code removal and new driver addition.